### PR TITLE
Add orga setting 'require_duplicate_from'

### LINKED ISSUE #2507: Meeting.create: make duplicate from (optionally) mandatory
Context: The new setting from the Meta-Issue https://github.com/OpenSlides/openslides-meta/issues/125

If the new setting is active then the action meeting.create requires values from „duplicate from“

### PR DESCRIPTION
**Pull request by reiterl**
Resolve #2507

### CODE REVIEW

**Comment by reiterl** [comment]
@luisa-beerboom  Please review.

**Comment by luisa-beerboom** [comment]
> Perhaps I need to forbid the meeting.create for committee admins, if the setting is set?

No, just throw a permission error, when a committee admin calls meeting.create without the `duplicate_from` field

**Review by luisa-beerboom — APPROVED** [review]

**Comment by Elblinator** [comment]
Some requirements were not documented and now need some implementing in this PR:
If the new setting is activated then....
- A committee admin should not be allowed to turn meetings into templates/ remove them as templates
- A committee admin should not be allowed to duplicate

**Comment by reiterl** [comment]
If a cm is not allowed to duplicate a meeting ("meeting.clone") they cannot create a new meeting, because the
create a meeting from a template is a meeting.clone.
I suppose, a cm can only clone template meetings if duplicate from is required. Or do I miss s.th. ?

**Comment by Elblinator** [comment]
> If a cm is not allowed to duplicate a meeting ("meeting.clone") they cannot create a new meeting, because the create a meeting from a template is a meeting.clone. I suppose, a cm can only clone template meetings if duplicate from is required. Or do I miss s.th. ?

If that is possible then yes, only allow a cm to clone template meetings if the settings is active.

**Comment by reiterl** [comment]
@luisa-beerboom I have add the two changes @Elblinator requested. (With docs and tests).
Please review.

File: docs/actions/meeting.update.md [hunk 1]
```diff
@@ -202,3 +202,4 @@ This action doesn't allow for a meeting to be set as a template and have `locked
 - Admins of the meeting can modify group D
 - Users with CML `can_manage` or users with a OML of `can_manage_organization` can modify group E
 - Only users with OML `superadmin` can modify group F
+- Users need OML `can_manage_organization` to `set_as_template` if organization setting `require_duplicate_from` is set.
```
Comment (luisa-beerboom) [inline docs/actions/meeting.update.md:205]: To keep the format, `set_as_template` should be it's own group

File: openslides_backend/action/actions/meeting/clone.py [hunk 2]
```diff
@@ -79,6 +82,21 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
...
+            ONE_ORGANIZATION_FQID, ["require_duplicate_from", "template_meeting_ids"]
+        )
+        if organization.get("require_duplicate_from"):
+            if not has_organization_management_level(
+                self.datastore,
+                self.user_id,
+                OrganizationManagementLevel.CAN_MANAGE_ORGANIZATION,
+            ) and not instance["meeting_id"] in (
+                organization.get("template_meeting_ids") or []
+            ):
```
Comment (luisa-beerboom) [inline openslides_backend/action/actions/meeting/clone.py:95]: Either combine the if-conditions using and (i.e. `organization.get("require_duplicate_from") and not has_organization_management_level(self.datastore, self.user_id, OrganizationManagementLevel.CAN_MANAGE_ORGANIZATION) and not instance["meeting_id"] in (organization.get("template_meeting_ids") or [])`) or do something like the following
```suggestion
        if not instance["meeting_id"] in (
            organization.get("template_meeting_ids") or []
        ):
            organization = self.datastore.get(
                ONE_ORGANIZATION_FQID, ["require_duplicate_from", "template_meeting_ids"], lock_result=False
            )
            if organization.get("require_duplicate_from") and not has_organization_management_level(
                self.datastore,
                self.user_id,
                OrganizationManagementLevel.CAN_MANAGE_ORGANIZATION,
            ):
```

File: openslides_backend/action/actions/meeting/clone.py [hunk 3]
```diff
@@ -79,6 +82,21 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
...
+                raise ActionException(
+                    "Committee manager cannot clone a non-template meeting, if duplicate-from is required."
```
Comment (luisa-beerboom) [inline openslides_backend/action/actions/meeting/clone.py:97]:
```suggestion
                    "Committee manager cannot clone a non-template meeting if duplicate-from is required."
```

File: openslides_backend/action/actions/meeting/update.py [hunk 4]
```diff
@@ -227,6 +228,19 @@ def update_instance(self, instance: dict[str, Any]) -> dict[str, Any]:
...
+        )
+        if organization.get("require_duplicate_from"):
+            if set_as_template is not None and not has_organization_management_level(
+                self.datastore,
+                self.user_id,
+                OrganizationManagementLevel.CAN_MANAGE_ORGANIZATION,
+            ):
+                raise ActionException(
+                    "A meeting cannot be set as a template by a committee manager, if duplicate from is required."
+                )
```
Comment (luisa-beerboom) [inline openslides_backend/action/actions/meeting/update.py:242]: see my comment in `meeting/clone.py`

**Review by luisa-beerboom — CHANGES_REQUESTED** [review]

File: docs/actions/meeting.update.md (hunk 1 above)
Comment (reiterl) [reply docs/actions/meeting.update.md:205]: Now use  a Group G for 'set_as_template'.

File: openslides_backend/action/actions/meeting/clone.py (hunk 2 above)
Comment (reiterl) [reply openslides_backend/action/actions/meeting/clone.py:95]: Use 'and' to combine the if-conditionals.

File: openslides_backend/action/actions/meeting/clone.py (hunk 3 above)
Comment (reiterl) [reply openslides_backend/action/actions/meeting/clone.py:97]: Rm the comma

File: openslides_backend/action/actions/meeting/update.py (hunk 4 above)
Comment (reiterl) [reply openslides_backend/action/actions/meeting/update.py:242]: Use 'and' to combine the if-conditionals and rm the comma.

**Comment by reiterl** [comment]
I have worked on the changes for the CR. I hope, this is now okay. @luisa-beerboom pls review.

**Review by luisa-beerboom — CHANGES_REQUESTED** [review]
the datastore calls should not lock fields

**Comment by reiterl** [comment]
Don't lock the fields of the two datastore calls, which get the organization.

**Review by luisa-beerboom — APPROVED** [review]

**Pull request merged**